record.go

```diff
@@ -0,0 +1,444 @@
+package libdns
+
+import (
+	"fmt"
+	"net/netip"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Record is any type that can reduce itself to the [RR] struct.
+type Record interface {
+	RR() (RR, error)
```

Comment:
Couldn't this be `RR() RR`. What could the error be? I've done a quick attempt to change dnsclay to use this new interface, and I have to do `rr, _ := record.RR()` everywhere. The Record-implementing types all have an `RR()` that always returns success. The RR.Parse function seems to be for parsing in a way that can return errors.

Reply:
I did at first, but then I was like "You know, not returning an error got me in trouble with a couple exported APIs when I designed Caddy, so it doesn't hurt to do so... I might as well here." That was before any RR struct types had been implemented.

Now that several RR types have been implemented, I think it's more obvious to me that they should always be able to reduce to an RR, without error. (That is, after all, the definition of a type that implements the Record interface. If it has an error, it's like... not even a DNS record??)

So yeah, maybe we can get rid of the error return value here. What do you think @gucci-on-fleek ?

Reply:
I went ahead and pushed a commit that makes it `RR() RR`, no error.

I _assume_ this will be okay, although now I remember one instance where an error WAS returned... when `type Address` used to be two separate types, `A` and `AAAA`, in an earlier revision, if the `netip.Addr` field wasn't the respective IPv4 or IPv6 bit length, then there would have been an error.

Without returning an error, I'm not sure what we'd have done. Just let it happen and expect the DNS provider API to return an error when trying to put an IPv6 address in an A record or something?

So, while it seems to be okay _currently_, I am super curious if there's ever a possibility that an error would be useful. If so, I'd like to restore it.

Reply:
> Couldn't this be `RR() RR`. What could the error be? I've done a quick attempt to change dnsclay to use this new interface, and I have to do `rr, _ := record.RR()` everywhere. 

Yeah, I also had to do `rr, _ := record.RR()`, so this is a :+1: from me.